presto-orc/src/main/java/com/facebook/presto/orc/OrcSelectiveRecordReader.java

```diff
@@ -737,10 +741,11 @@ else if (!hasAnyFilter(columnIndex)) {
             }
         }
 
+        if (appendRowNumber) {
+            blocks[outputColumns.size()] = createRowNumbersBlock(positions, positionCount, this.getFilePosition());
```

Comment:
Using positions seems incorrect, the above code looks like starts with position and adjusts the positions based on if filter functions are present. 

Can you also please add a test, to repro this issue and then use positionsToRead to fix this code.

Reply:
You are right, thanks for catching that. I added a test case to cover this